### PR TITLE
Dedicated Security Content

### PR DESCRIPTION
**Pull request by ahardin-rh**
Updated Preview Build:
http://file.rdu.redhat.com/~ahardin/04252019/dedicated-security/product_security/

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@jeremyeder @anjalitelang PTAL at my first pass at bringing the Security content into the Dedicated docs for 3.11. Thanks!

**Comment by jeremyeder** [comment]
Looks good. Understand you're restructuring a bit and will have another version shortly.

**Comment by ahardin-rh** [comment]
@jeremyeder @anjalitelang Thank you! My rework is now complete. Please review the latest preview build and let me know if any changes are needed:
http://file.rdu.redhat.com/~ahardin/04232019/dedicated-security/product_security/index.html

File: product_security/recommendations_for_customers/customer_logging_metrics_monitoring.adoc
Comment (anjalitelang) [inline product_security/recommendations_for_customers/customer_logging_metrics_monitoring.adoc]: Change to Audit logs

File: product_security/recommendations_for_customers/customer_logging_metrics_monitoring.adoc
Comment (anjalitelang) [inline product_security/recommendations_for_customers/customer_logging_metrics_monitoring.adoc]: Add text: Audit logs are maintained by Red Hat for Compliance purposes and are not shareable with customers at this time.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Add text and link: 
Learn more about Authentication in OpenShift Dedicated at : http://file.rdu.redhat.com/~ahardin/04232019/dedicated-security/architecture/additional_concepts/authentication.html

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Can we add this text right below the Authorization heading:

Authorization is managed using:
Rules : Sets of permitted verbs on a set of objects. For example, whether something can create pods.
Roles : Collections of rules. Users and groups can be associated with, or bound to, multiple roles at the same time.
Bindings: Associations between users and/or groups with a role.

OpenShift Dedicated evaluates authorizations using the following steps:

1. The identity and the project-scoped action is used to find all bindings that apply to the user or their groups.
2. Bindings are used to locate all the roles that apply.
3. Roles are used to find all the rules that apply.
4. The action is checked against each rule to find a match.
5. If no matching rule is found, the action is then denied by default.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Move this link above right after the text for Authorization

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: RBAC should be a sub-heading under "Authorization"

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Cluster roles and local roles Should be a sub-heading under RBAC.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Move the reference at the end of RBAC paragraph (Right before we start talking about SCC)

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Add text:
There are two levels of RBAC roles and bindings that control authorization:

Cluster RBAC : Roles and bindings that are applicable across all projects. Roles that exist cluster-wide are considered cluster roles. Cluster role bindings can only reference cluster roles.

Local RBAC : Roles and bindings that are scoped to a given project. Roles that exist only in a project are considered local roles. Local role bindings can reference both cluster and local roles.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Add link. 
For more information on the roles supported on the cluster see http://file.rdu.redhat.com/~ahardin/04232019/dedicated-security/architecture/additional_concepts/authorization.html#roles

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: SCC should be a sub-heading under Authorization

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Add text below SCC heading:
Security context constraints allow administrators to control permissions for pods. As an OpenShift Dedicated cluster administrator, you can list and view details for SCCs, but cannot edit or delete the default SCCs.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/access_control.adoc]: Red Hat Personnel Access should be under "Authorization" header

File: product_security/red_hat_responsibilities/dedicated_network_security.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/dedicated_network_security.adoc]: Add following below:
Additionally, some dedicated clusters also support the ovs-multitenant plug-in that provides project-level isolation for pods and services. Each project receives a unique Virtual Network ID (VNID) that identifies traffic from pods assigned to the project. By default, Pods from different projects cannot send packets to or receive packets from pods and services of a different project. However, the cluster administrator can join projects and later isolate them using admin commands. For more information on Pod network management see https://docs.openshift.com/dedicated/admin_guide/managing_networking.html

File: product_security/red_hat_responsibilities/dedicated_network_security.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/dedicated_network_security.adoc]: Add link: https://docs.openshift.com/container-platform/3.11/install_config/configuring_aws.html#configuring-a-security-group-aws

File: product_security/red_hat_responsibilities/dedicated_network_security.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/dedicated_network_security.adoc]: Add following text:
Security Groups provide security at the protocol and port access level. Each security group works similar to a firewall and has a set of rules that filter traffic coming into and out of a node.

File: product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc]: Needs a heading called Logging

File: product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc]: This is my bad. The Customer controlled monitoring needs to go under Recommendations for customers, with Logging.

File: product_security/responsibilities_of_cloud_providers/index.adoc
Comment (anjalitelang) [inline product_security/responsibilities_of_cloud_providers/index.adoc]: Add link to AWS security: https://aws.amazon.com/security/

File: product_security/recommendations_for_customers/restrict_access.adoc
Comment (anjalitelang) [inline product_security/recommendations_for_customers/restrict_access.adoc]: Rename to Service Serving Certificate Secrets - Application level PKI

File: product_security/recommendations_for_customers/secure_container_lifecycle_management.adoc
Comment (anjalitelang) [inline product_security/recommendations_for_customers/secure_container_lifecycle_management.adoc]: At the end of the whole page, maybe we can add this reference link https://www.redhat.com/en/resources/container-security-openshift-cloud-devops-whitepaper#

**Review by anjalitelang — CHANGES_REQUESTED** [review]
Hi Ashley,
Really great job in bringing this all together. I have a few comments about changing heading or adding links to some places. It should be good to go after that. Thank you again for your super effort on this!
Anjali.

File: product_security/recommendations_for_customers/restrict_access.adoc
Comment (anjalitelang) [inline product_security/recommendations_for_customers/restrict_access.adoc]: add after including route-specific annotations - " and IP whitelisting"

**Review by anjalitelang — CHANGES_REQUESTED** [review]
One more minor change.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (ahardin-rh) [reply product_security/red_hat_responsibilities/access_control.adoc]: Is it okay if I omit the text and just move the link up to the content in xref:../../architecture/additional_concepts/authorization.adoc#architecture-additional-concepts-authorization[Learn
how authorization is managed] instead since that text is pulled from there? I don't want to duplicate content in our doc set.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (ahardin-rh) [reply product_security/red_hat_responsibilities/access_control.adoc]: It is. This is an h3 heading while Authorization is an h2. The TOC reflects this, but I know the heading sizes are very similar and hard to distinguish at times.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (ahardin-rh) [reply product_security/red_hat_responsibilities/access_control.adoc]: It is. This is an h3 heading while Authorization is an h2. The TOC reflects this, but I know the heading sizes are very similar and hard to distinguish at times.

File: product_security/recommendations_for_customers/secure_container_lifecycle_management.adoc
Comment (ahardin-rh) [reply product_security/recommendations_for_customers/secure_container_lifecycle_management.adoc]: Good idea. I am adding an Additional Resources section with this link

File: product_security/recommendations_for_customers/restrict_access.adoc
Comment (ahardin-rh) [reply product_security/recommendations_for_customers/restrict_access.adoc]: Adding in the text, but I'll leave it unbolded per our style guidelines

**Comment by ahardin-rh** [comment]
@anjalitelang Thanks for your feedback! I addressed your comments. Please let me know if I missed anything. The updated preview build can be reviewed here:
http://file.rdu.redhat.com/~ahardin/04252019/dedicated-security/product_security/

@jeremyeder PTAL as well. Thanks!

**Review by anjalitelang — APPROVED** [review]
Looks great and thanks for the quick turnaround on the reviews from yesterday. Two very very minor changes from my end:
1. Can we rename "Customer-controlled Monitoring" to "Monitoring" in the customer responsibility Logging and Monitoring section.
2. I see that the two roles under RH responsibility->Access control->RBAC show up with all Caps - CLUSTER ROLES AND LOCAL ROLES. Can we change this to normal text. 

Thanks very much again for this! Please feel free to forward to: Shawn Purtell, John Golding, Dave Baker, Doug Edgar and Patrick Strick. Once we have ack from Jeremy and the team we can commit. Thanks again,
Anjali

**Comment by ahardin-rh** [comment]
@anjalitelang Thank YOU! I updated  "Customer-controlled Monitoring" to "Monitoring". Unfortunately, I can't address your second point, as this is our style for all h4 headings. Please let me know if anything else needs addressed.

@patrickstrick @jeremyeder @spurtell @jgouldin @dbaker-rh @rhdedgar Please review this content for the Dedicated 3.11 Security Guide. Thank you!

File: product_security/recommendations_for_customers/secure_container_lifecycle_management.adoc
Comment (anjalitelang) [inline product_security/recommendations_for_customers/secure_container_lifecycle_management.adoc]: Replace with - Container processes run under a UID assigned by the OpenShift Platform, and never with root privileges.

**Comment by anjalitelang** [comment]
@patrickstrick @jeremyeder @spurtell @jgouldin  @rhdedgar Please review this content for the Dedicated 3.11 Security Guide. Thank you!

File: product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc
Comment (patrickstrick) [inline product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc]: This section confuses me. What do we mean by "monitoring service"? The 2nd sentence then talks about logging. Then, the 3rd sentence says automated testing, config mgmt, etc. This seems like it is talking about three different things.

File: product_security/red_hat_responsibilities/compliance.adoc
Comment (patrickstrick) [inline product_security/red_hat_responsibilities/compliance.adoc]:
```suggestion
Certificaiton of Red Hat {product-title} on AWS against compliance standards such
```
?

File: product_security/red_hat_responsibilities/compliance.adoc
Comment (patrickstrick) [inline product_security/red_hat_responsibilities/compliance.adoc]: I am not sure we should use the language "ascertain" and "ensuring" as those may be legally binding. Check with legal first.

File: product_security/recommendations_for_customers/disaster_recovery.adoc
Comment (patrickstrick) [inline product_security/recommendations_for_customers/disaster_recovery.adoc]:
```suggestion
is the responsibility of the customer to backup application specific
```

File: product_security/recommendations_for_customers/disaster_recovery.adoc
Comment (patrickstrick) [inline product_security/recommendations_for_customers/disaster_recovery.adoc]:
```suggestion
data for example, special export and import procedures for database systems. In
```

**Review by patrickstrick — CHANGES_REQUESTED** [review]

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [reply product_security/red_hat_responsibilities/access_control.adoc]: We originally had Tier[x] added but removed it as it is too detailed for product docs.

File: product_security/red_hat_responsibilities/access_control.adoc
Comment (anjalitelang) [reply product_security/red_hat_responsibilities/access_control.adoc]: We originally had Tier[x] added but removed it as it is too detailed for product docs.

File: product_security/red_hat_responsibilities/dedicated_network_security.adoc
Comment (anjalitelang) [reply product_security/red_hat_responsibilities/dedicated_network_security.adoc]: +1

File: product_security/red_hat_responsibilities/dedicated_network_security.adoc
Comment (anjalitelang) [reply product_security/red_hat_responsibilities/dedicated_network_security.adoc]: +1 Patrick.  I think this just got left out in the comments and never got fixed. 
We default when I do openssl s_client -connect api.bu-demo.openshift.com:443, the cert offered is TLS1.2.. but yes I tested with TLS1.1/SSLv3 and TLS1.0 and it connected. Rephrasing to not include SSLv2/v3 reference and adding
The
{product-title} server and `oc` client provide TLS 1.2 or newer by default.

File: product_security/red_hat_responsibilities/dedicated_network_security.adoc
Comment (anjalitelang) [inline product_security/red_hat_responsibilities/dedicated_network_security.adoc]: Replace lines 73 to remove SSL reference.
Replace Line 74 with The
{product-title} server and `oc` client provides TLS 1.2 or newer by default.

File: product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc
Comment (anjalitelang) [reply product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc]: Background: I got this section from the Service definition mainly to highlight how we monitor the cluster. We did not want to use Zabbix reference so used the term centrallized logging instead for aggregation of the logs. 
Recommended Change:  
{product-title} clusters come with an integrated Prometheus/Grafana stack for cluster monitoring. Monitoring service containers runs on each application node and scrap data from the nodes which is then fronted on the Grafana dashboard. The aggregated data is further used by Red Hat SRE for services such as pruning, garbage collection, automated testing and configuration management.

File: product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc
Comment (anjalitelang) [reply product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc]: +1

**Review by anjalitelang — CHANGES_REQUESTED** [review]
Please see replies to Patrick

File: product_security/red_hat_responsibilities/compliance.adoc
Comment (ahardin-rh) [reply product_security/red_hat_responsibilities/compliance.adoc]: @patrickstrick Okay, thanks! I will try to track someone down.

Comment (ahardin-rh) [reply product_security/red_hat_responsibilities/compliance.adoc]: @patrickstrick @anjalitelang I opened a ticket with Legal and copied you. The rest of your feedback is now addressed. Thanks!

**Comment by ahardin-rh** [comment]
Per discussion with @patrickstrick and @anjalitelang, we removed a sentence that is currently being reviewed by Red Hat Legal. All other comments from @patrickstrick @anjalitelang and @dbaker-rh are addressed, so I am merging this. If Legal comes back with approval on the Compliance verbiage, we can add it back in via a separate PR.

**Pull request merged**
